app/src/services/cpk.ts

```diff
@@ -256,6 +263,56 @@ class CPKService {
       throw err
     }
   }
+
+  sellOutcomes = async ({
+    amount,
+    outcomeIndex,
+    marketMaker,
+    conditionalTokens,
+  }: CPKSellOutcomesParams): Promise<TransactionReceipt> => {
+    try {
+      const signer = this.provider.getSigner()
+      const account = await signer.getAddress()
+
+      const outcomeTokensToSell = await marketMaker.calcSellAmount(amount, outcomeIndex)
+      const collateralAddress = await marketMaker.getCollateralToken()
+
+      const transactions = [
+        {
+          operation: CPK.CALL,
+          to: conditionalTokens.address,
+          value: 0,
+          data: ConditionalTokenService.encodeSetApprovalForAll(marketMaker.address, true),
+        },
+        {
+          operation: CPK.CALL,
+          to: marketMaker.address,
+          value: 0,
+          data: MarketMakerService.encodeSell(amount, outcomeIndex, outcomeTokensToSell),
+        },
+        {
+          operation: CPK.CALL,
+          to: collateralAddress,
+          value: 0,
+          data: ERC20Service.encodeApproveUnlimited(account),
+        },
+        {
+          operation: CPK.CALL,
+          to: collateralAddress,
+          value: 0,
+          data: ERC20Service.encodeTransferFrom(this.cpk.address, account, amount),
```

Comment:
I don't know if this makes sense. The address that will call the `transferFrom` **is** the CPK, so these two transactions can be replaced with just a `transfer`:

```
{
          operation: CPK.CALL,
          to: collateralAddress,
          value: 0,
          data: ERC20Service.encodeTransfer(account, amount),
}

Reply:
I added the transfer method to the ERC20 service, and remove the the two transactiosn that you mention